include/dos_inc.h

```diff
@@ -292,19 +292,27 @@ class DOS_PSP :public MemStruct {
 
 	void	SaveVectors			(void);
 	void	RestoreVectors		(void);
-	void	SetSize				(Bit16u size)			{ sSave(sPSP,next_seg,size);		};
-	Bit16u	GetSize				(void)					{ return (Bit16u)sGet(sPSP,next_seg);		};
-	void	SetEnvironment		(Bit16u envseg)			{ sSave(sPSP,environment,envseg);	};
-	Bit16u	GetEnvironment		(void)					{ return (Bit16u)sGet(sPSP,environment);	};
-	Bit16u	GetSegment			(void)					{ return seg;						};
+
+	void SetSize(uint16_t size) { sSave(sPSP, next_seg, size); }
+	uint16_t GetSize() { return (uint16_t)sGet(sPSP, next_seg); }
+
+	void SetEnvironment(uint16_t eseg) { sSave(sPSP, environment, eseg); }
+	uint16_t GetEnvironment() { return (uint16_t)sGet(sPSP, environment); }
+
+	uint16_t GetSegment() { return seg; }
+
 	void	SetFileHandle		(Bit16u index, Bit8u handle);
 	Bit8u	GetFileHandle		(Bit16u index);
-	void	SetParent			(Bit16u parent)			{ sSave(sPSP,psp_parent,parent);	};
-	Bit16u	GetParent			(void)					{ return (Bit16u)sGet(sPSP,psp_parent);		};
-	void	SetStack			(RealPt stackpt)		{ sSave(sPSP,stack,stackpt);		};
-	RealPt	GetStack			(void)					{ return sGet(sPSP,stack);			};
-	void	SetInt22			(RealPt int22pt)		{ sSave(sPSP,int_22,int22pt);		};
-	RealPt	GetInt22			(void)					{ return sGet(sPSP,int_22);			};
+
+	void SetParent(uint16_t parent) { sSave(sPSP, psp_parent, parent); }
+	uint16_t GetParent() { return (uint16_t)sGet(sPSP, psp_parent); }
```

Comment:
Maybe static_cast

Reply:
I prefer to keep existing cast (for now); in reality there should be *no* cast in here at all; `sGet` is a mix of macros and static methods causing UB (detected by one of our sanitizers), and casting of result is already inconsistent (sometimes it's ommitted).

I have old WIP patch fixing it - reworking `sGet` to avoid UB; when I'll get to cleaning up that patch, I'll cover these pointless casts as well.